### PR TITLE
legacy split clone copy data bug fix

### PR DESCRIPTION
**Pull request by xhh1989**
legacy split clone copy data bug fix, which may lead data Inconsistent、duplicate Primary key.

### CODE REVIEW

**Comment by xhh1989** [comment]
I signed it!

File: go/vt/worker/legacy_split_clone.go [hunk 1]
```diff
@@ -567,7 +567,7 @@ func (scw *LegacySplitCloneWorker) copy(ctx context.Context) error {
 
 			for _, c := range chunks {
 				sourceWaitGroup.Add(1)
-				go func(td *tabletmanagerdatapb.TableDefinition, tableIndex int, chunk chunk) {
+				go func(td *tabletmanagerdatapb.TableDefinition, shardIdx, tableIndex int, chunk chunk) {
```
Comment (michael-berlin) [inline go/vt/worker/legacy_split_clone.go]: nit: Can you please the new parameter "shardIndex"?

For `tableIndex` we reuse the same name as well. This should work?

**Comment by michael-berlin** [comment]
Yeah, this is an actual issue :(

I'm surprised the e2e test didn't catch this: https://github.com/youtube/vitess/blob/master/test/merge_sharding.py

Do we have to change the test?

File: go/vt/worker/legacy_split_clone.go (hunk 1 above)
Comment (xhh1989) [inline go/vt/worker/legacy_split_clone.go]: done

**Pull request merged**